lenskit/algorithms/basic.py

```diff
@@ -153,6 +153,60 @@ def inverse_transform(self, ratings):
             rvps['rating'] += rvps['u_off'].fillna(0)
         return rvps.drop(columns=['u_off', 'i_off'])
 
+    def transform_user(self, ratings):
+        """
+        Transform a user's ratings by subtracting the bias model.
+
+        Args:
+            ratings(pandas.Series): The user's ratings, indexed by item.
+                Must have at least `item` as index and `rating` column.
+
+        Returns:
+            pandas.Series:
+                The transformed ratings and the user bias.
+        """
+        ratings = ratings.subtract(self.mean_)
+
+        # build user_offsets_
+        if self.item_offsets_ is not None:
+            rv = ratings.values - self.item_offsets_[
+                self.item_offsets_.index.isin(ratings.index)].values
+        else:
+            rv = ratings
+        user_offsets_ = self._mean(rv, self.user_damping)
+
+        # subtracts bias
+        if self.item_offsets_ is not None:
+            ratings = ratings.subtract(
+                    self.item_offsets_[self.item_offsets_.index.isin(ratings.index)].values)
+
+        ratings = ratings.subtract(user_offsets_)
+        return ratings, user_offsets_
+
+    def inverse_transform_user(self, user, ratings, user_bias=None):
+        """
+        Un-transform a user's ratings by adding in the bias model.
+
+        Args:
+            user: The user ID.
+            ratings(pandas.Series): The user's ratings, indexed by item.
+            user_bias(float or None): If `None`, it looks up the user bias learned by `fit`.
+
+        Returns:
+            pandas.Series: The user's de-normalized ratings.
+        """
+        ratings = ratings.add(self.mean_)
+        if self.item_offsets_ is not None:
+            ratings = ratings.add(self.item_offsets_[
+                self.item_offsets_.index.isin(ratings.index)].values)
+        if user_bias is not None:
+            ratings = ratings.add(user_bias)
+        elif self.user_offsets_ is not None:
+            current_u_b = self.user_offsets_[self.user_offsets_.index == user]
```

Comment:
You can do `self.user_offsets_.loc[user]` and it will be faster:

```python
elif self.user_offsets_ is not None and user in self.user_offsets_.index:
    ratings = ratings + self.user_offsets_.loc[user]
```